### PR TITLE
Use spring boot + JPA to deal with statistics dao

### LINKED ISSUE #838: Use spring boot + JPA to deal with statistics dao
Current statistics dao is using JDBC which can not adapt different database dialects.
It is better to use spring boot + JPA to more databases.

### PR DESCRIPTION
**Pull request by nmyphp**
Fixes #838.

Changes proposed in this pull request:
- Use spring boot + JPA refect RDBStatisticRepository
- Add DynamicDataSourceConfig for dynamic switch datasource

### CODE REVIEW

**Comment by Technoboy-** [comment]
Hi, @nmyphp .
From the CI log, the format or code style of some classes are not satisfied.
This can be checked by running "mvn clean install" locally.

**Review by terrymanu — CHANGES_REQUESTED** [review]
Please fix CI fail, the reason is for check style

**Comment by nmyphp** [comment]
> Please fix CI fail, the reason is for check style

sorry, fixed.

**Review by Mark-WJQ — COMMENTED** [review]
Should the time style  be TIMESTAMP? like  @Column(nullable = false, columnDefinition = "TIMESTAMP")

File: elastic-job-lite-console/src/main/java/org/apache/shardingsphere/elasticjob/lite/console/domain/TaskResultStatistics.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import lombok.NoArgsConstructor;
+
+/**
+ * Task result statistics.
+ */
+@Getter
+@AllArgsConstructor
+@NoArgsConstructor
+@Entity
+@Table(name = "TASK_RESULT_STATISTICS")
```
Comment (Mark-WJQ) [inline elastic-job-lite-console/src/main/java/org/apache/shardingsphere/elasticjob/lite/console/domain/TaskResultStatistics.java:40]: TASK_RESULT_STATISTICS  in the original project shuld be three tables task_result_statistics_minute,task_result_statistics_hour,task_result_statistics_day, now it merge to one?

**Comment by nmyphp** [comment]
> Should the time style be TIMESTAMP? like @column(nullable = false, columnDefinition = "TIMESTAMP")

Thank you, I will test it.

**Pull request merged**